Assets/MixedRealityToolkit/_Core/DataProviders/SpatialObservers/WindowsMixedReality/WindowsMixedRealitySpatialObserver.cs

```diff
@@ -121,33 +104,19 @@ public override void Destroy()
         /// one as a spare for use in outstanding mesh requests. That way, we'll have fewer
         /// game object create/destroy cycles, which should help performance.
         /// </summary>
-        protected SpatialMeshObject? spareMeshObject = null;
+        private SpatialMeshObject? spareMeshObject = null;
 
         /// <summary>
         /// The time at which the surface observer was last asked for updated data.
         /// </summary>
         private float lastUpdated = 0;
-#endif // UNITY_WSA
 
         /// <inheritdoc />
-        public override IDictionary<int, GameObject> Meshes
-        {
-            get
-            {
-                Dictionary<int, GameObject> meshes = new Dictionary<int, GameObject>();
-                // NOTE: We use foreach here since Dictionary<key, value>.Values is an IEnumerable.
-                foreach (int id in meshObjects.Keys)
-                {
-                    meshes.Add(id, meshObjects[id].GameObject);
-                }
-                return meshes;
-            }
-        }
+        public override IReadOnlyDictionary<int, SpatialMeshObject> Meshes => meshObjects;
```

Comment:
Okay, let me take another look at this.

Reply:
For 1: Are we sure that applications don't need access to the `SpatialMeshObject`?

What's an example of how an application could request them on demand?

For 2: This makes sense on why we want a copy.

Reply:
I was under the impression that the `IReadOnlyCollection` would provide an immutable copy, but after some research, sadly [it doesn't](https://visualstudiomagazine.com/articles/2012/08/07/new-read-only-collection-interfaces-for-net.aspx).

>One caveat to note is that the interfaces do not provide an immutable copy of the collection but rather a read-only view of the source mutable collection.

Reply:
>For 1: Are we sure that applications don't need access to the SpatialMeshObject?
What's an example of how an application could request them on demand?

Abbreviated example, if n app wants the renderer for a specific mesh:

List<GameObject> meshes = SpatialAwarenessSystem.Meshes;
…
meshes[i].GetComponent<Renderer>();

While this may not be the most performant mechanism, it protects the internal SpatialMeshObject (SMO) data from being modified by an app. (While today, SMO is defined as a struct, a new PR will make it a class).

Reply:
Yeah let's not use `getComponent` as it's expensive, esp for a large collection of objects.

Part of the problem is that the `SpatialMeshObject` is using unprotected fields instead of properties counter to the coding guidelines.